src/device-manager/java/src/nl/Weave/DataManagement/WdmClient.java

```diff
@@ -67,7 +67,7 @@
 
     public interface CompletionHandler
     {
-        void onFlushUpdateComplete();
+        void onFlushUpdateComplete(int[] pathResults);
```

Comment:
What are these results, exactly? It's unclear how this array should be interpreted.

Reply:
For example,
In your one flush operation,
        localSettingsTrait.set("/1", "en-US");
        testCTrait.set("/1", false);
        testCTrait.setSigned("/2", 15);
        testCTrait.setUnsigned("/3/1", -5);
        testCTrait.set("/3/2", false);
        testCTrait.setUnsigned("/4", -5);
        TestResult = null;
        mockWdmClient.beginFlushUpdate();
        ExpectSuccess("FlushUpdate");
If all succeeds, you pathResults would be 
{0, 0, 0, 0, 0, 0}, which is exactly matching with your flushed path, and it is ErrorList for your flushed path. 
If the first fails, you pathResults would be 
{4041, 0, 0, 0, 0, 0},  4041 is the invalid profile id error code, it would tell you the trait may not be implemented in device side.
In application, you need to record the list regarding trait and tag path, you can use deleteData to clear the first path, or you can use clear to clear the whole trait.

Reply:
What is the error code space? i.e. where is 4041 defined?

Reply:
it should be weave error code, in my standalone mock test, it starts from 4000